elasticsearch/docker/config/docker-entrypoint.sh

```diff
@@ -46,17 +45,39 @@ done < <(env)
 # will run in.
 export ES_JAVA_OPTS="-Des.cgroups.hierarchy.override=/ $ES_JAVA_OPTS"
 
-# TODO: Replace supervisord with something else so we can certify the docker image (https://docs.docker.com/docker-hub/publish/certify-images/ says supervisord is forbidden)
-#if [[ -d "/usr/share/elasticsearch/plugins/opendistro-performance-analyzer" ]]; then
-#    CLK_TCK=`/usr/bin/getconf CLK_TCK`
-#    ES_JAVA_OPTS="-Dclk.tck=$CLK_TCK -Djdk.attach.allowAttachSelf=true $ES_JAVA_OPTS"
-#    if [[ -d "/usr/share/elasticsearch/performance-analyzer-rca" ]]; then
-#        ES_JAVA_OPTS="-Djava.security.policy=file:///usr/share/elasticsearch/performance-analyzer-rca/pa_config/es_security.policy $ES_JAVA_OPTS"
-#        /usr/bin/supervisord -c /usr/share/elasticsearch/performance-analyzer-rca/pa_config/supervisord.conf
-#    else
-#        ES_JAVA_OPTS="-Djava.security.policy=file:///usr/share/elasticsearch/plugins/opendistro-performance-analyzer/pa_config/es_security.policy $ES_JAVA_OPTS"
-#        /usr/bin/supervisord -c /usr/share/elasticsearch/plugins/opendistro-performance-analyzer/pa_config/supervisord.conf
-#    fi
-#fi
-
-/usr/share/elasticsearch/bin/elasticsearch "${es_opts[@]}"
+
+# Start up the elasticsearch and performance analyzer agent processes.
+# When either of them halts, this script exits, or we receive a SIGTERM or SIGINT signal then we want to kill both these processes.
```

Comment:
It makes sense to kill the RCA process if elasticsearch halts, but I'm not sure if the opposite is true. If it's possible, I think it would make sense to try to restart the RCA process.

Reply:
I'd rather follow container best practice and just terminate the whole container. Let the container management software handle restarts.